src/main/groovy/edu/oregonstate/mist/cardsapi/db/CardDAO.groovy

```diff
@@ -71,8 +35,61 @@ interface CardDAO extends Closeable {
     """)
     Card getCardById(@Bind("id") Integer id)
 
+    @SqlUpdate ("""
+        INSERT INTO CARDS (ID, TYPE_ID, NAME, COLOR_ID, RARITY_ID, ENERGY, DESCRIPTION)
+        VALUES (
+            (:id),
+            (SELECT TYPE_ID FROM CARD_TYPES WHERE TYPE = :type),
+            (:name),
+            (SELECT COLOR_ID FROM CARD_COLORS WHERE COLOR = :color),
+            (SELECT RARITY_ID FROM CARD_RARITIES WHERE RARITY = :rarity),
+            (:energy),
+            (:description)
+        )
+        """)
+    void postCard(@Bind("id") Integer id,
+                  @Bind("type") String type,
+                  @Bind("name") String name,
+                  @Bind("color") String color,
+                  @Bind("rarity") String rarity,
+                  @Bind("energy") Integer energy,
+                  @Bind("description") String description)
+
+    @SqlQuery("SELECT CARD_INSTANCE_SEQ.NEXTVAL FROM DUAL")
+    Integer getNextId()
+
+    @SqlUpdate ("""
+        UPDATE CARDS
+            SET TYPE_ID = (SELECT TYPE_ID FROM CARD_TYPES WHERE TYPE = :type),
+            NAME = :name,
+            COLOR_ID = (SELECT COLOR_ID FROM CARD_COLORS WHERE COLOR = :color),
+            RARITY_ID = (SELECT RARITY_ID FROM CARD_RARITIES WHERE RARITY = :rarity),
+            ENERGY = :energy,
+            DESCRIPTION = :description
+        WHERE CARDS.ID = :id
+        
+    """)
+    void putCard(@Bind("id") Integer id,
+                 @Bind("type") String type,
+                 @Bind("name") String name,
+                 @Bind("color") String color,
+                 @Bind("rarity") String rarity,
+                 @Bind("energy") Integer energy,
+                 @Bind("description") String description)
+
+    // Check if card exists
+    @SqlQuery ("""
```

Comment:
You could change this into a Boolean method and use a query like `select count(*) from card where cards.id = :id`. The method will return true for any non-zero value. Then using this method could be simpler, you could just do `if (carddao.cardExists) {`